sdk/core/azure-core/tests/test_retry_policy.py

```diff
@@ -141,17 +143,19 @@ def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineRe
                     assert not position
                     return HttpResponse(request, None)
 
-    file_name = 'test_retry_seekable_file'
+    file_name = str(uuid.uuid4())
     with open(file_name, "w+") as f:
         f.write('Lots of dataaaa')
     http_request = HttpRequest('GET', 'http://127.0.0.1/')
     headers = {'Content-Type': "multipart/form-data"}
     http_request.headers = headers
-    form_data_content = {
-        'fileContent': open(file_name, 'rb'),
-        'fileName': file_name,
-    }
-    http_request.set_formdata_body(form_data_content)
-    http_retry = RetryPolicy(retry_total = 1)
-    pipeline = Pipeline(MockTransport(), [http_retry])
-    pipeline.run(http_request)
+    with open(file_name, 'rb') as f:
+        form_data_content = {
+            'fileContent': f,
+            'fileName': file_name,
+        }
+        http_request.set_formdata_body(form_data_content)
+        http_retry = RetryPolicy(retry_total=1)
+        pipeline = Pipeline(MockTransport(), [http_retry])
+        pipeline.run(http_request)
+    os.remove(file_name)
```

Comment:
Would [`with tempfile.NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) work here? If it does, it would clean up after itself

Reply:
If I understand correctly, tempfile will be removed automatically after close. Hence it will be gone after
    with open(file_name, "w+") as f:
        f.write('Lots of dataaaa')?

Reply:
Sure, but couldn't that write occur in the tempfile `with` block as the first action?

Reply:
This is a good question. And in fact I purposely separate them.  :)

My thought is 

file_name = str(uuid.uuid4())
    with open(file_name, "w+") as f:
        f.write('Lots of dataaaa')

are the test preparation steps and after that are real test scenario. I don't want to mix them though I don't think it will impact the results.

Reply:
I wish the docs were more clear about this. They state:
> Whether the name can be used to open the file a second time, *while the named temporary file is still open*, varies across platforms 

Which actually leads me to believe that using `NamedTemporaryFile` to create a file which you write data to *and then close*, would be OK to open again later. That might be preferable to saving a file with a UUID filename in the current dir. There are definitely cases where writing the file might fail (non-writable curdir) or deleting may fail (an exception in the test) leaving junk in the repo (which always risks being checked in by mistake). 

Which I guess raises another avenue: why not just check in a sample data file in to the test directory?

Reply:
tempfile has a delete parameter which is exactly what we want. update the case to use tempfile